core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java

```diff
@@ -321,7 +319,7 @@ public CompletableFuture<ConnectedRuleDetails> getActiveRuleDetails(EndpointPara
     var ruleDefFromPluginOpt = analysisContext.get().findRule(ruleKey);
     if (ruleDefFromPluginOpt.isPresent()) {
       var ruleDefFromPlugin = ruleDefFromPluginOpt.get();
-      if (globalConfig.getExtraPluginsPathsByKey().containsKey(ruleDefFromPlugin.getLanguage().getPluginKey()) || projectKey == null) {
```

Comment:
Removing this condition will cause an error when connected to an old SQ. Without the condition we will try to get the rule from the server while it does not exist. I suggest to keep loading the active rule details from the rule definition with the condition: rule language is secrets and serverConnection.supportSecretAnalysis() is false